src/js/files/components/UploadItem.js

```diff
@@ -39,3 +41,42 @@ export const UploadItem = ({ name, progress, size }) => (
         </UploadItemTitle>
     </StyledUploadItem>
 );
+
+const StyledUploadItemError = styled(StyledUploadItem)`
+    i.fa-times {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 20px;
+    }
+    span: last-child {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 12px;
+        font-style: italic;
+    }
+    i.fa-trash {
+        margin-left: 5px;
+        font-size: 14px;
+    }
+`;
+
+export const UploadItemError = ({ name, onRemove, localId }) => (
+    <StyledUploadItemError>
+        <AffixedProgressBar now={100} color="red" />
+        <UploadItemTitle>
+            {<Icon name="times" />}
+            {name}
+            <span>
+                Failed <Icon aria-label={`delete ${name}`} name="trash" color="red" onClick={() => onRemove(localId)} />
```

Comment:
Prime use case for `useCallback`.

Reply:
Hmm, honestly not convinced of the performance benefit here. My understanding of `useCallback` is that is best used in cases where referential equality matters. I could perhaps see it making a difference when react is diffing the dom tree, but most of the examples I have seen talking about efficient `useCallback` usage primarily suggested reserving it for cases where you needed the referential equality for proper functioning (ex lodash throttle) or to save massive downstream re-renders. (ex. when the function is a dependency for a large `useMemo`)

Icon technically does check referential equality when defining its own `useCallback` but looking closer I think this usage is a bit suspect to start:
```javascript
const handleClick = useCallback(props.onClick, [props.onClick]);
```
Since `useCallback` returns the original reference to the function there are two ways this can go. On a new render either the `props.onClick` has the exact same reference as before, and the usage of useCallback was redundant since the returned `handleClick` has the exact same reference as `props.onClick`, or `props.onClick` points to a new reference and useCallback now returns a reference identical to the new `props.onClick`.

Let me know what you think. Happy to change course if I have missed something.

Reply:
You're right. I commented on the pattern without thinking to deeply about it.